frames/frames.go

```diff
@@ -0,0 +1,119 @@
+// Package frames contains an API for interacting with i-frames and representing them into some data structure.
+package frames
+
+import (
+	"image"
+	"io"
+
+	// Due to the way the image package works, we need to register the png import as well, even if it isn't used directly
+	_ "image/png"
+)
+
+// CompressionType represents the type of Compression used on the video file.
+type CompressionType int
+
+const (
+	// MPEG2 is used for files saved with .mp2 and have i-frames every 0.5 seconds.
+	MPEG2 CompressionType = iota
+	// MPEG4 is used for files saved with .mp4 and have i-frames every 0.20002 seconds.
+	MPEG4
+)
+
+// Processor is an interface for processing i-frames into exported Frame structures. The interface requires a single
+// method, Process, which accepts a filename and returns a Frame type to represent it.
+type Processor interface {
+	Process(io.Reader) (uint32, uint32, uint32, uint32, error)
+}
+
+// Frame represents an i-frame by maintaining a single value for each color channel and the number of frame.
+type Frame struct {
+	num int    // The frame number within the video
+	r   uint32 // Red color channel value
+	g   uint32 // Green color channel value
+	b   uint32 // Blue color channel value
+	a   uint32 // Alpha blending factor value
+}
+
+// NewFrame creates a Frame structure.
+func NewFrame(num int, r, g, b, a uint32) Frame {
+	return Frame{
+		num: num,
+		r:   r,
+		g:   g,
+		b:   b,
+		a:   a,
+	}
+}
+
+// Red returns the value of the red color channel.
+func (f Frame) Red() uint32 {
+	return f.r
+}
+
+// Green returns the value of the green color channel.
+func (f Frame) Green() uint32 {
+	return f.g
+}
+
+// Blue returns the value of the blue color channel.
+func (f Frame) Blue() uint32 {
+	return f.b
+}
+
+// Num returns the number of the frame.
```

Comment:
Would it be better to refer the **number** of the frame to the **ID** of the frame?

Reply:
Yeah that's probably better. Good suggestion!